Setup periodic snyk monitoring per branch

breskeby opened the pull request:
Adds a new ci Jenkins job configuration for running snyk dependency monitoring on a daily basis. We setup a service account in snyk and resolve the api token for publishing in vault. 

Related to #87620

breskeby committed:
Setup periodic snyk monitoring per branch

mark-vieira commented on .ci/jobs.t/elastic+elasticsearch+periodic+snyk-dependency-monitoring.yml:
```diff
@@ -0,0 +1,23 @@
+          VAULT_TOKEN=$(vault write -field=token auth/approle/login role_id=$VAULT_ROLE_ID secret_id=$VAULT_SECRET_ID)
+          export VAULT_TOKEN
+          export data=$(vault read -format=json secret/elasticsearch-ci/snyk)
+          export snyk_token=$(echo $data | jq -r .data.token)
```
We can simplify this to `snyk_token=$(vault read -field=token secret/elasticsearch-ci/snyk)`.

breskeby replied:
neat. done

mark-vieira commented on .ci/jobs.t/elastic+elasticsearch+periodic+snyk-dependency-monitoring.yml:
```diff
@@ -0,0 +1,23 @@
+          export snyk_token=$(echo $data | jq -r .data.token)
+          unset VAULT_TOKEN data
+          set -x
+          $WORKSPACE/.ci/scripts/run-gradle.sh uploadSnykDependencyGraph -PsnykToken=$snyk_token
```
We can't do `set -x` and then include the token in the command line as this is going to leak the secret. We'll either need to tweak the build to pull this from an env var, or omit the `set -x` bit so we don't print this command. I'm inlined to do the former, as it's probably less likely that we'll make a mistake and accidentally leak the token in console output when using environment variables as we do for most other secrets.

breskeby committed:
Apply review feedback;

- use env variable instead of project property which allows not leaking token in ci environment
- simplify reading token from vault

mark-vieira reviewed (approved):
LGTM
